openmc/data/multipole.py

```diff
@@ -661,3 +651,47 @@ def __call__(self, E, T):
 
         fun = np.vectorize(lambda x: self._evaluate(x, T))
         return fun(E)
+
+    def export_to_hdf5(self, path, libver='earliest'):
+        """Export windowed multipole data to an HDF5 file.
+
+        Parameters
+        ----------
+        path : str
+            Path to write HDF5 file to
+        libver : {'earliest', 'latest'}
+            Compatibility mode for the HDF5 file. 'latest' will produce files
+            that are less backwards compatible but have performance benefits.
+
+        """
+
+        # Open file and write version.
+        with h5py.File(path, 'w', libver=libver) as f:
+            f.create_dataset('version', (1, ), dtype='S10')
+            f['version'][:] = WMP_VERSION.encode('ASCII')
+
+            # Make a nuclide group.
+            g = f.create_group('nuclide')
+
+            # Write scalars.
+            if self.formalism == 'MLBW':
+                g.create_dataset('formalism',
+                                 data=np.array(_FORM_MLBW, dtype=np.int32))
+            else:
+                # Assume RM.
+                g.create_dataset('formalism',
+                                 data=np.array(_FORM_RM, dtype=np.int32))
+            g.create_dataset('spacing', data=np.array(self.spacing))
+            g.create_dataset('sqrtAWR', data=np.array(self.sqrtAWR))
+            g.create_dataset('start_E', data=np.array(self.start_E))
+            g.create_dataset('end_E', data=np.array(self.end_E))
+
+            # Write arrays.
+            g.create_dataset('data', data=self.data)
+            g.create_dataset('l_value', data=self.l_value)
+            g.create_dataset('pseudo_K0RS', data=self.pseudo_k0RS)
+            g.create_dataset('w_start', data=self.w_start)
+            g.create_dataset('w_end', data=self.w_end)
+            g.create_dataset('broaden_poly',
+                             data=self.broaden_poly.astype(np.int8))
```

Comment:
Wouldn't this actually confuse the Fortran side (which is essentially expecting a 32-bit `int`)?

Reply:
HDF5 does type conversion internally.